File: php/src/Creational/Builder/BostonSoupBuffetBuilder.php
<?

#//BostonSoupBuffetBuilder.java - One of Two Builder Subclasses

require_once 'SoupBuffetBuilder.php';

class BostonSoupBuffetBuilder extends SoupBuffetBuilder
{
	function buildClamChowder() { $this->soupBuffet->clamChowder = new BostonClamChowder() ; }
	function buildFishChowder() { $this->soupBuffet->fishChowder = new BostonFishChowder() ; }
	function setSoupBuffetName() { $this->soupBuffet->soupBuffetName = "Boston Soup Buffet"; }
}

File: python/src/Creational/Builder/BostonSoupBuffetBuilder.py

#/BostonSoupBuffetBuilder - Two of Two Builder Subclasses

from SoupBuffetBuilder import SoupBuffetBuilder
from common.BostonClamChowder import BostonClamChowder
from common.BostonFishChowder import BostonFishChowder

class BostonSoupBuffetBuilder(SoupBuffetBuilder):
	def __init__(self) : pass
	def buildClamChowder(self)   : 
		self.soupBuffet.clamChowder = BostonClamChowder()
	def buildFishChowder(self)   : 
		self.soupBuffet.fishChowder = BostonFishChowder()
	def setSoupBuffetName(self) : 
		self.soupBuffet.soupBuffetName = "Boston Soup Buffet"
